_posts/2024-01-30-securing-wildfly-apps-oidc-k8s.adoc

```diff
@@ -0,0 +1,398 @@
+* A Kubernetes cluster.
++
+You can use a local Kubernetes cluster such as link:https://minikube.sigs.k8s.io/docs/[minikube], or https://kind.sigs.k8s.io/[kind].
++
+The steps provided in this guide use minikube.
+
+* Kubernetes command-line tool, https://kubernetes.io/docs/tasks/tools/[kubectl].
+
+* A public container registry to host your application's Docker image.
++
+For example, https://hub.docker.com/[Docker
+Hub].
+* https://helm.sh/docs/intro/install/[Helm Chart].
+
+
+== Example Application
+
+We use a simple web application in this guide that consists of a single https://github.com/wildfly-security-incubator/elytron-examples/blob/main/simple-webapp-oidc/src/main/java/org/wildfly/security/examples/SecuredServlet.java[servlet]. We show how to secure this servlet using OIDC. We will use the example in the https://github.com/wildfly-security-incubator/elytron-examples/tree/main/simple-webapp-oidc[simple-webapp-oidc] directory in the https://github.com/wildfly-security-incubator/elytron-examples[elytron-examples] repository.
+
+To obtain this example, clone the elytron-examples repository to your local machine:
+
+[source]
+----
+git clone git@github.com:wildfly-security-incubator/elytron-examples.git
+----
+
+== Start Keycloak
+
+We will be using Keycloak as our OpenID provider.
+
+Follow the instructions, up until "Log in to the Admin Console", provided in the https://www.keycloak.org/getting-started/getting-started-kube[Get started with Keycloak on Kubernetes] guide.
+
+== Configure Keycloak
+
+To create a realm in Keycloak, add a user and role to the realm, and create a client with which to secure your application, follow these steps:
+
+. Log into the Keycloak Admin Console using the username and password you specified earlier.
+
+. Create a new realm called *myrealm*. For more information, see the Keycloak documentation on how to https://www.keycloak.org/getting-started/getting-started-kube#_create_a_realm[create a realm].
+
+. Add a role called *user*. This role will be required to access our simple web application. For more information, see the Keycloak documentation on how to https://www.keycloak.org/docs/latest/server_admin/index.html#assigning-permissions-using-roles-and-groups[create a role].
+
+. Add a new user named *alice*. Set an *email* address for this new user, we'll use *alice@example.org*. For more information, see the Keycloak documentation on how to https://www.keycloak.org/getting-started/getting-started-kube#_create_a_user[create a user].
+
+. Once the new user has been created, set a password for this new user from the *Credentials* tab.
+
+. From the *Role Mapping* tab, assign *alice* the *user* role. For more information, see the Keycloak documentation on how to https://www.keycloak.org/docs/latest/server_admin/index.html#proc-assigning-role-mappings_server_administration_guide[assign a role] to a user.
+
+. Create a new client as follows:
+* *General Settings*:
```

Comment:
These directions are more concise and clear

Click the General Settings image in the top left hand corner of the admin console
Click the "Clients" menu item
Click the "Create client" button
On the "Create client" page set the "Client type" to "OpenID Connect".
Set the "Client id" as, myclient.
Click the "Next" button at the bottom of the page
On the "Capability config" page the checkboxes for "Standard flow" 
and "Direct access grants" should be selected.
Click the "Next" button at the bottom of the page
On the "Login settings" page no action is needed at this time.
Click the "Save" button at the bottom of the page.

Reply:
agreed, updated with slight modifications.